src/Controller/Frontend.php

```diff
@@ -254,7 +254,25 @@ public function listing(Request $request, $contenttypeslug)
             $amount = empty($contenttype['listing_records']) ? $this->getOption('general/listing_records') : $contenttype['listing_records'];
         }
         if (!$order = $this->getOption('theme/listing_sort', false)) {
-            $order = empty($contenttype['sort']) ? $this->getOption('general/listing_sort') : $contenttype['sort'];
+            $order = empty($contenttype['sort']) ? null : $contenttype['sort'];
+        }
+        // If $order is not set, one of two things can happen: Either we let `getContent()` sort by itself, or we
+        // explicitly set it to sort on the general/listing_sort setting.
+        if ($order === null) {
+            $taxonomies = $this->getOption('taxonomy');
+            $hassortorder = false;
+            if (!empty($contenttype['taxonomy'])) {
+                foreach($contenttype['taxonomy'] as $contenttypetaxonomy) {
+                    if ($taxonomies[ $contenttypetaxonomy ]['has_sortorder']) {
+                        // We have a taxonomy with a sortorder, so we must keep $order = false, in order
+                        // to let `getContent()` handle it. We skip the fallback that's a few lines below.
+                        $hassortorder = true;
```

Comment:
> Tired and probably wrong… again!… but shouldn't we break; here?

I tried that at first, but couldn't get it to work.. `break;` only breaks out the inner loop. `break 2;` gives this error message: 

```
Fatal error: Cannot break/continue 2 levels in /Users/bob/Sites/bolt-origin/src/Controller/Frontend.php on line 271
Cannot break/continue 2 levels
```

So, i gave up, and did it like this. If you can get it to work, That'd be cleaner than this solution.